### PR TITLE
improve pickle tests for callbacks

### PR DESCRIPTION
**Pull request by jeremyjordan**
# Before submitting

- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to update the docs?   
- [x] Did you write any new necessary tests?  
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## What does this PR do?
Follow-on for #1636 

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/callbacks/early_stopping.py [hunk 1]
```diff
@@ -57,17 +61,20 @@ def __init__(self, monitor: str = 'val_loss', min_delta: float = 0.0, patience:
...
-            'auto': torch.gt if 'acc' in self.monitor else torch.lt
-        }
 
         if mode not in self.mode_dict:
             if self.verbose > 0:
                 log.info(f'EarlyStopping mode {mode} is unknown, fallback to auto mode.')
             self.mode = 'auto'
 
+        if self.mode == 'auto':
+            if self.monitor == 'acc':
```
Comment (Borda) [inline pytorch_lightning/callbacks/early_stopping.py:71]: I guess that there are more metrics which are auto max, not only acc, for example, precision, f1...
maybe add a todo to be updated with new metrics package

Comment (jeremyjordan) [reply pytorch_lightning/callbacks/early_stopping.py:71]: fwiw i think we should just ask the user explicitly (as a compromise i at least added logging to alert the user what we're doing)

Comment (Borda) [reply pytorch_lightning/callbacks/early_stopping.py:71]: so you mean completely drop `auto`?

Comment (jeremyjordan) [reply pytorch_lightning/callbacks/early_stopping.py:71]: yeah, although i wouldn't want to drop it if it'd be disruptive to our users

**Pull request merged**